src/main/java/org/broadinstitute/hellbender/tools/exome/CopyNumberTriStateSegmentCaller.java

```diff
@@ -0,0 +1,362 @@
+     * <p>Values between this
+     * constant and 0.0 as considered as a 0.0.
+     * </p>
+     * Values above this threshold
+     * are considered to indicate a log probability calculation problem that is
+     * worth to be reported as an error or warning.
+     */
+    private static final double MAX_LOG_PROB = 1e-3;
+
+    /**
+     * Cached value of ln(10)^-1
+     */
+    protected static final double INV_LN_10 = 1.0 / Math.log(10);
+
+    /**
+     * Maximum reportable output quality score; higher quality scores will
+     * capped to this value.
+     */
+    public static final double MAX_QUAL_SCORE = 199.99999999999;
+
+
+    public static final double PHRED_SCORE_PRECISION = .1;
+
+    public static final double LOG10_PROB_PRECISION = PHRED_SCORE_PRECISION * .1;
+
+    @ArgumentCollection
+    protected CopyNumberTriStateHiddenMarkovModelArgumentCollection modelArguments =
+            new CopyNumberTriStateHiddenMarkovModelArgumentCollection();
+
+    /**
+     * Z-scores transformation dimensions.
+     */
+    public enum ZScoreDimension {
+
+        /**
+         * Input coverage must be transformed in z-scores sample by sample.
+         */
+        SAMPLE,
+
+        /**
+         * Input coverage must be transformed in z-scores target by target.
+         */
+        TARGET,
+
+        /**
+         * No transformation must be performed.
+         */
+        NONE
+    }
+
+    @Argument(
+            doc = "In what dimension apply the zscore transformation",
+            fullName = ZSCORE_DIMENSION_FULL_NAME,
+            shortName = ZSCORE_DIMENSION_SHORT_NAME,
+            optional = false
+    )
+    protected ZScoreDimension zscoreDimension = ZScoreDimension.SAMPLE;
+
+    @Argument(
+            doc = "Input counts",
+            fullName = StandardArgumentDefinitions.INPUT_LONG_NAME,
+            shortName = StandardArgumentDefinitions.INPUT_SHORT_NAME,
+            optional = false
+    )
+    protected File inputFile;
+
+    @Argument(
+            doc = "Output file",
+            fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME,
+            shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME,
+            optional = false
+    )
+    protected File outputFile;
+
+    @ArgumentCollection
+    protected final TargetArgumentCollection targetArguments = new TargetArgumentCollection(() -> inputFile);
+
+    @Override
+    protected final Object doWork() {
+        final CopyNumberTriStateHiddenMarkovModel model = modelArguments.createModel();
+        final TargetCollection<Target> targets = targetArguments.readTargetCollection(false);
+        final ReadCountCollection inputCounts = readAndSortCountsByTargetCoordinates(targets);
+        applyZScoreTransformation(inputCounts);
+        checkForMissingTargetsInInputCounts(targets, inputCounts);
+        openOutput(outputFile, model, targets, inputCounts);
+        final Object result = makeCalls(model, targets, inputCounts);
```

Comment:
`makeCalls` always returns `SUCCESS` as long as it doesn't throw an exception, right?  So why not make `makeCalls` a `void` method and let `doWork` return `SUCCESS`?